README.md

```diff
@@ -59,136 +59,77 @@ First change the directory to the repo's root path:
-  # propagation (aboz lemma): none.
-  # propagation (aboz lemma): none.
-  # checking: (x1 y1), sat.
-# final unknown set #<set: 1 2 3>.
-# weak uniqueness: unsafe.
-# ./benchmarks/circomlib-cff5ab6/Decoder@multiplexer.r1cs is underconstrained. Below is a counterexample:
-  # inputs:
-    # m1.main.inp: 0
-  # first possible outputs:
-    # m1.main.out[0]: 0
-    # m1.main.out[1]: 0
-    # m1.main.success: 0
-  # second possible outputs:
-    # m2.main.out[0]: 1
-    # m2.main.out[1]: 0
-    # m2.main.success: 1
+The circuit is underconstrained
+Counterexample:
+  inputs:
+    main.inp: 0
+  first possible outputs:
+    main.out[0]: 1
+    main.out[1]: 0
+    main.success: 1
+  second possible outputs:
+    main.out[0]: 0
+    main.out[1]: 0
+    main.success: 0
+  first internal variables:
+    no first internal variables
+  second internal variables:
+    no second internal variables
+Exiting Picus with the code 9
 ```
 
 If you see this, it means the environment that you are operating on is configured successfully.
 
 ## Reusability Instructions
 
-### Quick Problem Solving for New Circuits/Benchmarks
-
-We also provide easy API to compile and solve for any new benchmarks created. First you can use the following script to compile arbitrary benchmark:
-
-```bash
-./picus-compile.sh <path-to-your-circom-file>
-```
-
-This will generate a `*.r1cs` file in the same path as your provided `*.circom` file. Then, use the following script to solve for the benchmark:
-
-```bash
-./picus-solve.sh <path-to-your-r1cs-file>
-```
-
-This will automatically invoke the tool and output the result.
-
 ### More Options and APIs of the Tool
 
 The following lists out all available options for running the tool.
 
 ```bash
-usage: picus.rkt [ <option> ... ]
+usage: picus.rkt [ <option> ... ] <source>
+  <source> must be a file with .circom or .r1cs extension
 
 <option> is one of
 
-/ --r1cs <p-r1cs>
-|    path to target r1cs
-| --circom <p-circom>
-\    path to target circom (need circom compiler in PATH)
+  --json
+     enable json logging (default: false)
   --noclean
      do not clean up temporary files (default: false)
   --patch-circom
-     patch circom file to add public inputs (only applicable for --circom, default: false)
+     patch circom file to add public inputs (only applicable for circom source, default: false)
   --opt-level <p-opt-level>
-     optimization level for circom compilation (only applicable for --circom, default: 0)
+     optimization level for circom compilation (only applicable for circom source, default: 0)
   --timeout <p-timeout>
      timeout for every small query (default: 5000ms)
   --solver <p-solver>
-     solver to use: z3 | cvc4 | cvc5 (default: z3)
+     solver to use: z3 | cvc4 | cvc5 (default: cvc5)
   --selector <p-selector>
      selector to use: first | counter (default: counter)
   --precondition <p-precondition>
-     path to precondition json (default: null)
+     path to precondition json (default: none)
```

Comment:
What is precondition meant for? If it isn't needed then let's get rid of it in another PR.